modules/hub/main.tf

```diff
@@ -19,28 +19,45 @@ locals {
 
   is_gke_flag               = var.use_kubeconfig ? 0 : 1
   create_cmd_gke_entrypoint = "${path.module}/scripts/gke_hub_registration.sh"
-  create_cmd_gke_body       = "${local.is_gke_flag} ${var.gke_hub_membership_name} ${var.location} ${var.cluster_name} ${local.gke_hub_sa_key} ${var.project_id} ${var.labels}"
+  create_cmd_gke_body       = "${local.is_gke_flag} ${var.gke_hub_membership_name} ${var.location} ${var.cluster_name} ${local.gke_hub_sa_key} ${var.project_id} ${var.hub_project_id == "" ? var.project_id : var.hub_project_id} ${var.labels}"
   destroy_gke_entrypoint    = "${path.module}/scripts/gke_hub_unregister.sh"
-  destroy_gke_body          = "${local.is_gke_flag} ${var.gke_hub_membership_name} ${var.location} ${var.cluster_name} ${var.project_id}"
+  destroy_gke_body          = "${local.is_gke_flag} ${var.gke_hub_membership_name} ${var.location} ${var.cluster_name} ${var.project_id} ${var.hub_project_id == "" ? var.project_id : var.hub_project_id}"
 }
 
 data "google_client_config" "default" {
 }
 
+data "google_project" "project" {
+  project_id = var.hub_project_id == "" ? var.project_id : var.hub_project_id
+}
+
 resource "google_service_account" "gke_hub_sa" {
   count        = var.use_existing_sa ? 0 : 1
   account_id   = var.gke_hub_sa_name
-  project      = var.project_id
+  project      = var.hub_project_id == "" ? var.project_id : var.hub_project_id
   display_name = "Service Account for GKE Hub Registration"
 }
 
 resource "google_project_iam_member" "gke_hub_member" {
   count   = var.use_existing_sa ? 0 : 1
-  project = var.project_id
+  project = var.hub_project_id == "" ? var.project_id : var.hub_project_id
   role    = "roles/gkehub.connect"
   member  = "serviceAccount:${google_service_account.gke_hub_sa[0].email}"
 }
 
+resource "google_project_iam_member" "hub_service_agent_project" {
+  project = var.project_id
+  role    = "roles/gkehub.serviceAgent"
+  member  = "serviceAccount:service-${data.google_project.project.number}@gcp-sa-gkehub.iam.gserviceaccount.com"
+}
+
+resource "google_project_iam_member" "hub_service_agent_hub" {
+  count   = var.hub_project_id == "" ? 0 : 1
+  project = var.hub_project_id
+  role    = "roles/gkehub.serviceAgent"
+  member  = "serviceAccount:service-${data.google_project.project.number}@gcp-sa-gkehub.iam.gserviceaccount.com"
+}
+
```

Comment:
IIUC when `hub_project_id ` is set we need to grant `roles/gkehub.serviceAgent` on `var.project_id`(GKE Project). This looks like we are granting across both projects?

Reply:
That is correct, I fixed this in the commit below. I removed adding the role to the local project and instead make sure the sa is enabled by using `google_project_service_identity`